README.md

```diff
@@ -6,12 +6,120 @@
  
 [npm-img]: https://img.shields.io/npm/v/ember-frost-list.svg "NPM Version"
 [npm-url]: https://www.npmjs.com/package/ember-frost-list
+
+[![Travis][ci-img]][ci-url] [![Coveralls][cov-img]][cov-url] [![NPM][npm-img]][npm-url]
+ 
+# ember-frost-list
+`ember-frost-list` is an ember addon that built on top of [smoke-and-miror](https://github.com/runspired/smoke-and-mirrors) which focuses on high performance list rendering. This component itself provides features such as list selection control, infinite scroll and list sorting. Consumer can take advantage of these features with simply providing the corresponding actions/functions 
  
-# ember-frost-list <br /> [![Travis][ci-img]][ci-url] [![Coveralls][cov-img]][cov-url] [![NPM][npm-img]][npm-url]
+ * [Installation](#Installation)
+ * [API](#API)
+ * [Examples](#Examples)
+ * [Contributing](#Contributing)
+
+## Installation
+```
+ember install ember-frost-list
+```
 
 ## API
 
-Coming soon
+## Basic usage
+| parameters type | Attribute | Type | Value | Description |
+|---------| --------- | ---- | ----- | ----------- |
+| `Positional param` | `N/A` | `string` | | path for user provided list-item component which describes each row/item in the list|
+| `Attribute` | `records` | `array` | | data model used for rendering list |
+| `Attribute` | `selections` | `array` | | array of currently selected items in list |
+| `Attribute` | `defaultHeight` | `number` | | default height for each list item, set to 45px if not provided |
+| `Attribute` | `onSelect` | `<action-name>` | | triggers associated user action when item in list gets selected |
+
+### Infinite scroll
+| parameters type | Attribute | Type | Description |
+|---------| --------- | ---- | ----------- |
+| `Attribute` | `loadPrevious` | `<action-name>` | triggers associated user action when scroll reaches upper boundary |
+| `Attribute` | `loadNext` | `<action-name>` | triggers associated user action when scroll reaches lower boundary |
+
+## Examples
+### Basic list
+In order to follow the data down action up patten, `ember-frost-list` holds no state or information of current selected list items. When an item selection event emits, The user provided actions will be called and attrs will be provided to consumer as first param of the function with all useful information that describe the current select event
+
+In template
+```handlebars
+{{frost-list
+  componentPath
+  records=listItems
+  selections=selectedItems
+  onSelect=(action 'selected')
+}}
+```
+In controller
+```javascript
+componentPath: 'user-list-item',
+listItems: Ember.computed.alias('model'),
+selectedItems: [],
+actions: {
+  selected (attrs) {
```

Comment:
As discussed, we want to move selectedItems and the selection handling into a utility (or possibly a mixin)

I'm leaning towards a mixin because it makes sense to wire up the selections to query params - on the other hand mixins aren't very good with multiple lists

Maybe a util that sets up everything we need during an init phase?